bounded-collections/src/bounded_btree_set.rs

```diff
@@ -359,22 +359,28 @@ where
 macro_rules! codec_impl {
 	($codec:ident) => {
 		use super::*;
-		use $codec::{Compact, Decode, DecodeLength, Encode, EncodeLike, Error, Input, MaxEncodedLen};
+		use crate::codec_utils::PrependCompactInput;
+		use $codec::{
+			Compact, Decode, DecodeLength, DecodeWithMemTracking, Encode, EncodeLike, Error, Input, MaxEncodedLen,
+		};
+
 		impl<T, S> Decode for BoundedBTreeSet<T, S>
 		where
 			T: Decode + Ord,
 			S: Get<u32>,
 		{
 			fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
-				// Same as the underlying implementation for `Decode` on `BTreeSet`, except we fail early if
-				// the len is too big.
-				let len: u32 = <Compact<u32>>::decode(input)?.into();
-				if len > S::get() {
+				// Fail early if the len is too big. This is a compact u32 which we will later put back.
+				let compact = <Compact<u32>>::decode(input)?;
```

Comment:
Nit: I would rename this to `len` since it holds the len of the collection, even though it's represented as a `Compact`. Also the same for `BTreeMap`.